### DISCUSSION TITLE
How to set filesystem paths for non-native plugins in .NET 8 Azure Functions?

### DISCUSSION THREAD

**Discussion by dluc** [microsoft/semantic-kernel#6571] [Q&A] [answered]
From Vorian on Discord:

> Anyone deploy SK to a .net 8 azure function with nom native function plugins  ? I can’t quite figure out how to set the path for reading files from file system ? Previously you could use the execution context and FunctionAppDirectory to set path to load plugins but now with .net 8 you don’t seem to have access to that due to  only isolated context for Azure functions ? 

src: https://discord.com/channels/1063152441819942922/1075907964667437072/1247978216979955732

**Comment by sophialagerkranspandey** [comment, marked as answer]
The pattern we recommend folks use is to use the Yaml format for prompt functions and load them as embedded resources. Here's a sample
[semantic-kernel/dotnet/samples/GettingStarted/Step3_Yaml_Prompt.cs at main · microsoft/semantic-kernel (github.com)](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/GettingStarted/Step3_Yaml_Prompt.cs#L28)
